src/components/sdk/MeetingControls/AudioInputControl.tsx

```diff
@@ -33,20 +33,32 @@ const AudioInputControl: React.FC<Props> = ({
   const selectAudioInput = useSelectAudioInputDevice();
   const { muted, toggleMute } = useToggleLocalMute();
   const { devices, selectedDevice } = useAudioInputs();
+  const [dropdownOptions, setDropdownOptions] = useState<PopOverItemProps[]>(
+    []
+  );
+
+  useEffect(() => {
```

Comment:
I do not think you need a state or useEffect here. If `devices` or `selectedDevice` changes this component needs to re-render anyhow right? So, can we keep this as is like earlier?

Reply:
No. useEffect is need here cause isOptionActive API is async now. And we can not use async function in React functional component.

Reply:
Also, with moving this handleClick, we have to update the dependency array with the variables, methods such as selectAudioInput. I wish we enabled the exhaustive-deps rule and solve this for us.

Reply:
Also, with moving this handleClick, we have to update the dependency array with the variables, methods such as selectAudioInput. I wish we enabled the exhaustive-deps rule and solve this for us.

Reply:
updated dependencies.